swift-ci/master/debian/10/Dockerfile

```diff
@@ -0,0 +1,38 @@
+FROM debian:10
+
+RUN groupadd -g 998 build-user && \
+    useradd -m -r -u 998 -g build-user build-user
+
+ENV DEBIAN_FRONTEND="noninteractive"
+
+RUN apt-get update
+
+RUN apt-get -y --no-install-recommends install \
+  build-essential       \
+  clang                 \
+  cmake                 \
+  git                   \
+  icu-devtools          \
+  libcurl4-openssl-dev  \
+  libedit-dev           \
+  libicu-dev            \
+  libncurses5-dev       \
+  libsqlite3-dev        \
+  libxml2-dev           \
+  ninja-build           \
+  python                \
+  python-dev            \
+  python-six            \
+  pkg-config            \
+  rsync                 \
+  swig                  \
+  systemtap-sdt-dev     \
+  tzdata                \
+  unzip                 \
+  vim                   \
+  uuid-dev		\
+  ca-certificates
```

Comment:
thanks @schafdog this is awesome. did you review all the above to make sure it the minimal set. for example I am a tad surprised to see `vim`

Reply:
Sorry, not at all optimized to be minimal. There was already a debian 9 dockerfile that I based 10 upon. Doing a manual build in the container I saw that just doing git clone failed due to missing CA certificates, which I added. I believe it did build after that. 

Still a bit insecure on how to "prove" that the dockerfile / image would work 100% to build the full toolchain and run the tests. Kind of interested in taking it all the way to debian packages, but can't say I have much experience in this.

Reply:
adding guide for local testing in https://github.com/apple/swift-docker/pull/171